src/tbb/arena.cpp

```diff
@@ -757,8 +760,11 @@ void task_arena_impl::execute(d1::task_arena_base& ta, d1::delegate_base& d) {
 
     bool same_arena = td->my_arena == a;
     std::size_t index1 = td->my_arena_index;
+    __TBB_ASSERT(td->my_worker_slot_type != thread_data::slot_type::undefined, "Invalid slot type");
+    bool is_worker_slot = td->my_worker_slot_type == thread_data::slot_type::worker;
     if (!same_arena) {
-        index1 = a->occupy_free_slot</*as_worker */false>(*td);
+        index1 = a->occupy_free_slot(*td);
+        is_worker_slot = false;
```

Comment:
Why we unset the flag?
I did not get the idea: how we distinguish if external thread occupied the workers slot.

Reply:
Good point, here can be a problem place. Logic behind `is_worker_slot = false;` is that the calling thread got `index1` slot as a master, so not consuming from quota of workers slots. Is it the invalid reasoning?

Reply:
What if there are more external threads than reserved slots then these external threads will take slots from workers quota.
Am I missing something?